### PR TITLE
Counselor authentication

### LINKED ISSUE #13: store and send passwords in a secure way
Plaintext is unnacceptable for storing passwords. It's important that integrity of counsel is guaranteed as much as possible. Some form of salted hashing should be used. Implementations likely already exist in libraries

### LINKED ISSUE #38: Add administrative abilities to server CLI
Counselors are currently only registered inside of the client, with the admin password being embedded and hardcoded to allow this functionality.

Hyperdome server should have command line arguments, and or configuration files to manage enrolled counselors.

This will also help for testing purposes, as adding a counselor currently either requires the hardcoded hack used in the client, or use of an API tester such as postman.

In the future a web based console (#3) is the ideal management interface

### PR DESCRIPTION
**Pull request by arades79**
### Summary

This PR finally authenticates counselors signing in!

This also finishes one CLI mechanism for adding counselors.

In this version, `hyperdome_server generate ` will generate a one-time-use code that can be given to a counselor. This code is placed in the "password" field when adding a server.
The client will authenticate using ed448 keys. Currently the counselor's keys are encrypted, but with a static hard-coded password.

Server settings are now stored and reloaded with the application.

This PR fixes #38  and invalidates #13 by cementing key authentication, and never broadcasting a password. 
### Checklist

- [x] If code changes were made then they have been tested.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: hyperdome/client/add_server_dialog.py [hunk 1]
```diff
@@ -123,14 +124,29 @@ def add_server(self):
         self.add_server_button.setText("Checking...")
 
         probe = threads.ProbeServerTask(self.session, self.server)
-        probe.signals.success.connect(self.set_server)
+        if self.server.is_counselor:
+            probe.signals.success.connect(self.signup)
+        else:
+            probe.signals.success.connect(self.set_server)
```
Comment (stevenpitts) [inline hyperdome/client/add_server_dialog.py:130]: Ternary is cool tho

File: hyperdome/client/main.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+
+class Application(QtWidgets.QApplication):
+    """
+    This is Qt's QApplication class. It has been overridden to support threads
+    and the quick keyboard shortcut.
+    """
+
+    def __init__(self, common):
+        if platform_str == "Linux" or platform_str == "BSD":
```
Comment (stevenpitts) [inline hyperdome/client/main.py:40]: I'm thinking this is old code, but
`if platform_str in ["Linux", "BSD"]:`

File: hyperdome/common/common.py [hunk 3]
```diff
@@ -31,8 +31,56 @@
...
+        # Look for resources directory relative to python file
+        prefix = os.path.join(
+            os.path.abspath(inspect.getfile(inspect.currentframe())), "share",
+        )
+        if not os.path.exists(prefix):
+            # While running tests during stdeb bdist_deb, look 3
+            # directories up for the share folder
+            prefix = os.path.join(
+                os.path.dirname(
+                    os.path.dirname(os.path.dirname(os.path.dirname(prefix)))
```
Comment (stevenpitts) [inline hyperdome/common/common.py:58]: I can't wait for the pathlib update

File: hyperdome/common/common.py [hunk 4]
```diff
@@ -31,8 +31,56 @@
...
+                ),
+                "share",
+            )
+
+    elif platform_str == "BSD" or platform_str == "Linux":
```
Comment (stevenpitts) [inline hyperdome/common/common.py:63]: Same as previous comment about platform_str

File: hyperdome/common/common.py [hunk 5]
```diff
@@ -42,13 +90,10 @@ class Common(object):
     def __init__(self, debug=False):
         self.debug = debug
 
-        # The platform OnionShare is running on
-        self.platform = platform.system()
-        if self.platform.endswith("BSD"):
-            self.platform = "BSD"
+        # The platform_str OnionShare is running on
```
Comment (stevenpitts) [inline hyperdome/common/common.py:93]: What's the comment for if the code is gone?
Also, Hyperdome*

File: hyperdome/common/types.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    @functools.wraps(fn)
+    def converted(*args, **kwargs):
+        args = [(arg.encode() if isinstance(arg, str) else arg) for arg in args]
+        kwargs = {
+            key: (value.encode() if isinstance(value, str) else value)
+            for key, value in kwargs.items()
+        }
+        return fn(*args, **kwargs)
+
+    return converted
```
Comment (stevenpitts) [inline hyperdome/common/types.py:41]: I love it ^_^

**Review by stevenpitts — APPROVED** [review]
Love it!

File: hyperdome/client/add_server_dialog.py (hunk 1 above)
Comment (arades79) [reply hyperdome/client/add_server_dialog.py:130]: ooh yeah, I had multiple lines in those for a couple commits, didn't think to refactor

File: hyperdome/client/main.py (hunk 2 above)
Comment (arades79) [reply hyperdome/client/main.py:40]: old code that I want to replace with pathlib in the near future

File: hyperdome/common/common.py (hunk 3 above)
Comment (arades79) [reply hyperdome/common/common.py:58]: same this sucks

File: hyperdome/common/types.py (hunk 6 above)
Comment (arades79) [reply hyperdome/common/types.py:41]: hell yeah, way better than having those lines at the top of 12 different functions

File: hyperdome/common/common.py (hunk 5 above)
Comment (arades79) [reply hyperdome/common/common.py:93]: gotta leave something to do next PR.
I should really just find+replace all references to OnionShare and micahflee at this point, I don't think there's any files left untouched.

**Pull request merged**